src/vea/vea_api.c

```diff
@@ -53,6 +56,29 @@ vea_upgrade(struct vea_space_info *vsi, struct umem_instance *umem,
 	if (rc != 0)
 		return rc;
 
+	/*
+	 * bitmap tree reused vec tree which was created with
+	 * BTR_FEAT_DIRECT_KEY, recreate tree with BTR_FEAT_UINT_KEY
```

Comment:
I think we could keep using BTR_FEAT_DIRECT_KEY for the bitmap tree.

Reply:
I did not do that originally, because we used md bitmap tree to store vea_hint_df, which could not use direct key, we might do that by modifying to add a wrapper vea_hint_df like:

struct vea_hint_df_direct {
           uint64_t key;
           struct vea_hint_df;
}

Reply:
Looks not working, direct key tree required fixed size value btree class? I hit problem locally:
tree EMRG src/common/btree_class.c:1585 ifv_rec_update() Assertion 'r->ir_value_len == val->iov_len' failed: Misuse of fixed size value btree class 24 != 48

Reply:
I see, I overlooked that vea_hint_df is stored in this tree with a special integer key.

Reply:
Not sure what is meant by "direct key tree required fixed size value btree class".  You can use direct key with any class that is registered to support direct key.